src/org/exoplatform/ui/setting/ServerEditionActivity.java

```diff
@@ -53,357 +52,326 @@
+    if (listAccounts.size() == 1) {
+      ExoAccount account = listAccounts.get(0);
+      AccountSetting settings = AccountSetting.getInstance();
+      settings.setCurrentAccount(account);
+      settings.setDomainIndex(String.valueOf(0));
+      Log.i(TAG, "Last account selected: " + account.accountName);
     }
-
-    /**
-     * Return to setting screen
-     */
-    private void returnToSetting(int operation, int serverIdx) {
-        Intent next = new Intent(this, SettingActivity.class);
-        next.setFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP | Intent.FLAG_ACTIVITY_CLEAR_TOP);
-        next.putExtra(ExoConstants.SETTING_TYPE,
-                      mIntent.getIntExtra(ExoConstants.SETTING_TYPE, SettingActivity.GLOBAL_TYPE));
-        next.putExtra(SETTING_OPERATION, operation);
-        next.putExtra(SERVER_IDX, serverIdx);
-        startActivity(next);
-    }
-
-    private View.OnClickListener onDeleteServer() {
-        return new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-
-                // decrease the selected account domain by 1 if the deleted
-                // account is before in the list
-                int selectedServerIndex = Integer.parseInt(mSetting.getDomainIndex());
-                if (mServerIdx < selectedServerIndex)
-                    mSetting.setDomainIndex(String.valueOf(selectedServerIndex - 1));
-
-                List<ExoAccount> listServer = ServerSettingHelper.getInstance()
-                                                                 .getServerInfoList(ServerEditionActivity.this);
-                listServer.remove(mServerIdx);
-
-                // remove current account setting if the last account was just
-                // deleted
-                if (listServer.isEmpty()) {
-                    mSetting.setCurrentAccount(null);
-                } else {
-                    // otherwise, check if there is only one remaining account
-                    // and if yes select it
-                    autoSelectLastAccount();
-                }
-                onSave();
-                Toast.makeText(ServerEditionActivity.this,
-                               mResources.getString(R.string.ServerDeleted),
-                               Toast.LENGTH_SHORT).show();
-                returnToSetting(SETTING_DELETE, mServerIdx);
-            }
-        };
+  }
+
+  /**
+   * Check whether account information is valid
+   * 
+   * @param myServerObj
+   * @return
+   */
+  private boolean isServerValid(ExoAccount myServerObj) {
+    boolean isValid = true;
+
+    // Account name and server are mandatory
+    if (myServerObj.accountName == null || myServerObj.serverUrl == null || myServerObj.accountName.length() == 0
+        || myServerObj.serverUrl.length() == 0) {
+      Toast.makeText(this, mResources.getString(R.string.WarningServerNameIsEmpty), Toast.LENGTH_LONG).show();
+      isValid = false;
```

Comment:
should return right after check server isValid = false to skip to unnecessary check